### PR TITLE
Adding publish code coverage  results task

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Added publish code coverage results task in preview.
Made changes to ant, maven and gradle to correct inconsistencies
Testing done: Manual testing.

### CODE REVIEW

File: Tasks/PublishCodeCoverageResults/task.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
+    "id": "2A7EBC54-C13E-490E-81A5-D7561AB7CD97",
+    "name": "PublishCodeCoverageResults",
+    "friendlyName": "Publish Code Coverage Results",
+    "description": "Publish code coverage results to VSO/TFS",
+    "helpMarkDown": "[More Information](http://go.microsoft.com/fwlink/?LinkID=613718)",
```
Comment (ajay-arora) [inline Tasks/PublishCodeCoverageResults/task.json]: is this correct ? Does it open the place holder readme file ?

File: Tasks/PublishCodeCoverageResults/task.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    "version": {
+        "Major": 1,
+        "Minor": 0,
+        "Patch": 0
+    },
+    "demands" : [
+    ],
+    "minimumAgentVersion": "1.88.0",
+    "groups": [        
+    ],
```
Comment (ajay-arora) [inline Tasks/PublishCodeCoverageResults/task.json]: we should remove this off

**Comment by ajay-arora** [comment]
looks fine.

File: Tasks/PublishCodeCoverageResults/task.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            "type": "string",
+            "label": "Report Directory",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Path of code coverage report directory. For example, $(Build.Repository.LocalPath)\\jacoco"
+        },
+        {
+            "name": "additionalCodeCoverageFiles",
+            "type": "string",
+            "label": "Additional files",
```
Comment (allendm-msft) [inline Tasks/PublishCodeCoverageResults/task.json]: capital F.. be consistent with the camel casing for all labels

File: Tasks/PublishCodeCoverageResults/task.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    "minimumAgentVersion": "1.88.0",
+    "groups": [        
+    ],
+    "instanceNameFormat": "Publish Code Coverage Results $(summaryFileLocation)",
+    "inputs": [
+        {
+            "name": "codeCoverageTool",
+            "type": "pickList",
+            "label": "Code Coverage Tool",
+            "required": false,
```
Comment (allendm-msft) [inline Tasks/PublishCodeCoverageResults/task.json]: shouldn't this be required?

File: Tasks/PublishCodeCoverageResults/PublishCodeCoverageResults.ps1 [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+Write-Verbose "codeCoverageTool = $codeCoverageTool" -Verbose
+Write-Verbose "summaryFileLocation = $summaryFileLocation" -Verbose
+Write-Verbose "reportDirectory = $reportDirectory" -Verbose
+Write-Verbose "additionalCodeCoverageFiles = $additionalCodeCoverageFiles" -Verbose
+
+# Import the Task.CodeCoverage dll that has all the cmdlets we need for Build
+import-module "Microsoft.TeamFoundation.DistributedTask.Task.CodeCoverage"
+
+# Publish Code Coverage Files
+$CodeCoverageFiles = $null
```
Comment (allendm-msft) [inline Tasks/PublishCodeCoverageResults/PublishCodeCoverageResults.ps1]: fix the variable name casing

File: Tasks/ANTPreview/Strings/resources.resjson/en-US/resources.resjson [hunk 6]
```diff
@@ -2,9 +2,9 @@
 // GENERATED FILE - DO NOT EDIT DIRECTLY
 // *******************************************************
 {
-  "loc.friendlyName": "Ant With CodeCoverage",
+  "loc.friendlyName": "Ant With Code Coverage",
```
Comment (allendm-msft) [inline Tasks/ANTPreview/Strings/resources.resjson/en-US/resources.resjson]: thanks for proactively fixing this

can you also make sure that all the labels in these tasks follow the camel casing

**Comment by allendm-msft** [comment]
signed off with suggestions

File: Tasks/ANTPreview/Strings/resources.resjson/en-US/resources.resjson (hunk 6 above)
Comment (AshwiniChalla) [inline Tasks/ANTPreview/Strings/resources.resjson/en-US/resources.resjson]: done.

**Pull request merged**